urllib3/connectionpool.py

```diff
@@ -99,6 +99,30 @@ def close(self):
 _blocking_errnos = set([errno.EAGAIN, errno.EWOULDBLOCK])
 
 
+# The buffering argument to HTTPConnection.getresponse is only supported
+# as an implementation detail of CPython 2.7, so other versions and other
+# implementations don't support it.
+def _get_httplib_response(conn):
+    """Get HTTP connection response with buffering explicitly enabled"""
+    try:
+        # CPython 2.7 supports this call, while other versions and
+        # implementations may not do so (e.g. 3.x doesn't)
+        return conn.getresponse(buffering=True)
+    except TypeError:
+        pass
```

Comment:
Rather than `except...pass` it might be better just to bring the rest of the code inside the except block.